### PR TITLE
Bottom drawer bringup

### PR DESCRIPTION
**Pull request by Renzo-Olivares**
This pull request includes the initial bringup of the bottom drawer for the reply study.
Includes the following changes:
* Bottom drawer that slides in from beneath the bottom app bar
* Current drawer destination is highlighted
* Bottom app bar has tappable logo
* Animated drop arrow next to logo
* Animated page title next to logo
* Main content opaque when bottom drawer slides out (drawer shade)
* Drawer retracts when touching drawer shade
* Migrate to using adaptive.dart instead of custom solution

Android Reply|Flutter Reply
---|---
android-b-reply|reply-bottom-initial

### CODE REVIEW

File: lib/studies/reply/bottom_drawer.dart [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return GestureDetector(
+      behavior: HitTestBehavior.opaque,
+      onVerticalDragUpdate: onVerticalDragUpdate,
+      onVerticalDragEnd: onVerticalDragEnd,
+      child: Material(
+        color: ReplyColors.blue700,
+        borderRadius: const BorderRadius.only(
+          topLeft: Radius.circular(12.0),
```
Comment (rami-a) [inline lib/studies/reply/bottom_drawer.dart]: Drop the`.0` everywhere for consitency.

File: lib/studies/reply/bottom_drawer.dart [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+              child: Text(
+                'FOLDERS',
+                style: Theme.of(context)
+                    .textTheme
+                    .caption
+                    .copyWith(color: ReplyColors.blue200),
+              ),
+            ),
+            const SizedBox(height: 4),
+            trailing
```
Comment (rami-a) [inline lib/studies/reply/bottom_drawer.dart]: Nit: trailing comma

File: lib/studies/reply/bottom_drawer.dart [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      child: Material(
+        color: ReplyColors.blue700,
+        borderRadius: const BorderRadius.only(
+          topLeft: Radius.circular(12.0),
+          topRight: Radius.circular(12.0),
+        ),
+        child: ListView(
+          padding: const EdgeInsets.all(12),
+          physics: const NeverScrollableScrollPhysics(),
+          children: <Widget>[
```
Comment (rami-a) [inline lib/studies/reply/bottom_drawer.dart]: Drop the `<Widget>`

File: lib/studies/reply/inbox.dart [hunk 4]
```diff
@@ -278,31 +286,339 @@ class _BuildDesktopNavState extends State<_BuildDesktopNav>
...
+      _drawerController.fling(
+          velocity: math.min(-_kFlingVelocity, -flingVelocity));
+    } else {
+      if (_drawerController.value < 0.6) {
+        _dropArrowController.forward();
+      }
+      _drawerController.fling(
+          velocity: _drawerController.value < 0.6
+              ? -_kFlingVelocity
+              : _kFlingVelocity);
```
Comment (rami-a) [inline lib/studies/reply/inbox.dart]: Nit: trailing comma and reformat

File: lib/studies/reply/inbox.dart [hunk 5]
```diff
@@ -278,31 +286,339 @@ class _BuildDesktopNavState extends State<_BuildDesktopNav>
...
+    final flingVelocity =
+        details.velocity.pixelsPerSecond.dy / _bottomDrawerHeight;
+
+    if (flingVelocity < 0.0) {
+      _drawerController.fling(
+          velocity: math.max(_kFlingVelocity, -flingVelocity));
+    } else if (flingVelocity > 0.0) {
+      _dropArrowController.forward();
+      _drawerController.fling(
+          velocity: math.min(-_kFlingVelocity, -flingVelocity));
```
Comment (rami-a) [inline lib/studies/reply/inbox.dart]: Nit: trailing comma and reformat

File: lib/studies/reply/inbox.dart [hunk 6]
```diff
@@ -278,31 +286,339 @@ class _BuildDesktopNavState extends State<_BuildDesktopNav>
...
+        _drawerController.status == AnimationStatus.completed) {
+      return;
+    }
+
+    final flingVelocity =
+        details.velocity.pixelsPerSecond.dy / _bottomDrawerHeight;
+
+    if (flingVelocity < 0.0) {
+      _drawerController.fling(
+          velocity: math.max(_kFlingVelocity, -flingVelocity));
```
Comment (rami-a) [inline lib/studies/reply/inbox.dart]: Nit: trailing comma and reformat

File: lib/studies/reply/inbox.dart [hunk 7]
```diff
@@ -278,31 +286,339 @@ class _BuildDesktopNavState extends State<_BuildDesktopNav>
...
+
+    final drawerAnimation = RelativeRectTween(
+      begin: RelativeRect.fromLTRB(0.0, drawerTop, 0.0, 0.0),
+      end: const RelativeRect.fromLTRB(0.0, 0.0, 0.0, 0.0),
+    ).animate(_drawerController.view);
+
+    return Stack(
+      overflow: Overflow.visible,
+      key: _bottomDrawerKey,
+      children: <Widget>[
```
Comment (rami-a) [inline lib/studies/reply/inbox.dart]: Nit: Drop `<Widget>`

**Review by rami-a — APPROVED** [review]
LGTM

File: lib/studies/reply/inbox.dart [hunk 8]
```diff
@@ -278,31 +290,342 @@ class _BuildDesktopNavState extends State<_BuildDesktopNav>
...
+            //Reload state when drawer is at its smallest to toggle visibility
+            //If state is reloaded before this drawer closes abruptly instead
+            //of animating.
+          });
+        }
+      });
+
+    _dropArrowController = AnimationController(
+        duration: const Duration(milliseconds: 350), vsync: this)
+      ..addListener(() {});
```
Comment (clocksmith) [inline lib/studies/reply/inbox.dart]: what is this listener for? Does not look like it is needed

File: lib/studies/reply/inbox.dart [hunk 9]
```diff
@@ -278,31 +290,342 @@ class _BuildDesktopNavState extends State<_BuildDesktopNav>
...
+
+  bool get _bottomDrawerVisible {
+    final status = _drawerController.status;
+    return status == AnimationStatus.completed ||
+        status == AnimationStatus.forward;
+  }
+
+  void _toggleBottomDrawerVisibility() {
+    if (_drawerController.value < 0.6) {
+      setState(() {});
```
Comment (clocksmith) [inline lib/studies/reply/inbox.dart]: What is this setState call for?

**Review by clocksmith — APPROVED** [review]

File: lib/studies/reply/inbox.dart (hunk 8 above)
Comment (Renzo-Olivares) [reply lib/studies/reply/inbox.dart]: Thanks I missed this.

File: lib/studies/reply/inbox.dart (hunk 9 above)
Comment (Renzo-Olivares) [reply lib/studies/reply/inbox.dart]: Nothing I forgot to remove it.

**Pull request merged**